cellmlmanip/parser.py

```diff
@@ -75,8 +75,7 @@ def set_parent(self, parent_name):
 
     def add_sibling(self, sibling_name):
         """Adds a sibling for this component"""
-        if sibling_name in self.siblings:
-            raise ValueError('Sibling component %s already added!' % sibling_name)
+        assert sibling_name not in self.siblings, 'Sibling component %s already added!' % sibling_name
```

Comment:
This doesn't seem to be reachable, when I tried to create it I got earlier errors about the variable already being assigned to.

Reply:
I confirmed, like this:

```

    def parse(self, cellml, wrap=True):
        """Writes some CellML to a temporary file and parses it."""
        if wrap:
            cellml = (
                '<?xml version="1.0" encoding="UTF-8"?>'
                '<model name="test" xmlns="http://www.cellml.org/cellml/1.0#">'
                + cellml +
                '</model>'
            )

        with tempfile.TemporaryDirectory() as d:
            path = os.path.join(d, 'file.cellml')
            with open(path, 'w') as f:
                f.write(cellml)
            return parser.Parser(path).parse()

    def test_encapsulation(self):
        """ Tests """

        # Parse valid sibling relationship
        self.parse(
            '<component name="a" />'
            '<component name="b" />'
            '<component name="c" />'
            '<group>'
            ' <relationship_ref relationship="encapsulation"/>'
            ' <component_ref component="a">'
            '  <component_ref component="b" />'
            '  <component_ref component="c" />'
            ' </component_ref>'
            '</group>'
        )
        # Note: this isn't stored, so can't check if it worked via public API!

        # Parse invalid relationship
        self.parse(
            '<component name="a" />'
            '<component name="b" />'
            '<component name="c" />'
            '<group>'
            ' <relationship_ref relationship="encapsulation"/>'
            ' <component_ref component="a">'
            '  <component_ref component="b" />'
            '  <component_ref component="c" />'
            ' </component_ref>'
            '</group>'
            '<group>'
            ' <relationship_ref relationship="encapsulation"/>'
            ' <component_ref component="a">'
            '  <component_ref component="b" />'
            '  <component_ref component="c" />'
            ' </component_ref>'
            '</group>'
        )
```

:D 

E           ValueError: Encapsulated component b already added!